test/aria/widgets/form/timefield/checkValue/TimeField.js

```diff
@@ -19,16 +19,115 @@ Aria.classDefinition({
-            time2 : null
-
+            time1 : null
         };
+        this.currentValue;
+        this.testValues = [];
+        this.checkValues = [];
+        this.separators = ["/", "\\", "h", ";", ",", ".", "-", ":"];
+        this.am = ["am", "a.m.", "a m"];
+        this.pm = ["pm", "p.m.", "p m"];
+        this.start();
         this.setTestEnv({
             data : this.data
         });
     },
     $prototype : {
+        start : function () {
+            this.check1Digit();
+            this.check2Digits();
+            this.check3Digits();
+            this.check4Digits();
+            this.checkAlphabetic();
+            this.checkValuesWithAMPM();
+        },
+        check1Digit : function () {
+            var hour, result, i;
+
+            // 1 digit = [h]
+            for (i = 0; i < 10; i++) {
+                hour = i + '';
+                result = '0' + i + ':00';
+                this.testValues.push(hour);
+                this.checkValues.push(result);
+            }
+        },
+        check2Digits : function () {
+            var hour, result, i;
+
+            // 2 digits = [h][h] (works with any of the separators after the first digit only)
+            for (i = 24; i > 22; i--) {
+                hour = i + '';
+                result = i !== 24 ? hour + ':00' : hour;
+                this.testValues.push(hour);
+                this.checkValues.push(result);
+            }
+            this.checkValuesWithSeparator(hour);
+        },
+        check3Digits : function () {
+            var hour, result;
+            // 3 digits = [h][h][m] (works with any of the separators after the first/second digit only)
+            hour = '150';
+            result = '15:00';
+            this.testValues.push(hour);
+            this.checkValues.push(result);
+            this.checkValuesWithSeparator(hour);
+        },
+        check4Digits : function () {
+            // 4 digits = [h][h]:[m][m] (works with any of the separators after the second digit only)
+            var value = "0000";
+            var result = "00:00";
+            this.testValues.push(value);
+            this.checkValues.push(result);
+            this.checkValuesWithSeparator(value);
+        },
+        checkAlphabetic : function () {
+            // alphabetic characters are invalid
+            this.testValues.push("12b", "b12", "1xx1", "xx");
+            this.checkValues.push("12b", "b12", "1xx1", "xx");
+
+        },
+        checkValuesWithAMPM : function () {
+            var valid = 1;
+            for (var items = 0; items < this.am.length; items++) {
+                // valid AM
+                this.testValues.push(valid + this.am[items]);
+                this.checkValues.push("01:00");
+
+                // valid PM
+                this.testValues.push(valid + this.pm[items]);
+                this.checkValues.push("13:00");
+            }
+        },
+        checkValuesWithSeparator : function (value) {
```

Comment:
You could lighten the code by using the square brackets notation instead of the `charAt` method. Of course in this case don't forget to convert the value to an array before to remain compatible with IE7: `value.split('')`.